### DISCUSSION TITLE
Network PCI Passthrough and SR-IOV

### DISCUSSION THREAD

**Discussion by stalhood** [apache/cloudstack#9963] [Q&A]
Are there steps out there for enabling PCI passthrough or SR-IOV for network cards to a VM in CloudStack 4.19? I've done GPUs already, but not finding a simple UI path for attaching a host NIC to a VM via the UI. Coming from VMware, having this capability for virtualized network functions to operate in CloudStack is important.

**Comment by DaanHoogland** [comment]
@stalhood , I think not, but I am not entirely sure of your requirement. Do you want to dedicate a host-NIC to a VM? i.e. no other VM will have access for it and it will not be used for any physical cloud network either.

**Comment by pcfriek1987** [comment]
@DaanHoogland With SR-IOV you split your physical nic in "functions", meaning that it will act as if there are more nics in your system, so that you can give an instance one of those functions so it will not use any of the hosts networking.

So you build it like you normally would, but next to that you can give instances their own nic.

**Comment by rajujith** [comment]
@stalhood It looks like there are no UI options yet. I am assuming you have used agent hooks (https://github.com/apache/cloudstack/pull/3839) for GPU.

**Comment by DaanHoogland** [comment]
closed this for lack of interaction, please reopen or open a new item if applicable
